core/src/main/java/uk/gov/gchq/koryphe/tuple/TupleInputAdapter.java

```diff
@@ -80,11 +81,7 @@ public FI apply(final Tuple<R> input) {
      */
     @SuppressFBWarnings(value = "EI_EXPOSE_REP2", justification = "Cloning the array would be expensive - we will have to reply on users not modifying the array")
     public void setSelection(final R[] selection) {
-        if (null == selection) {
-            this.selection = (R[]) new Object[0];
-        } else {
-            this.selection = selection;
-        }
+        this.selection = Objects.requireNonNullElseGet(selection, () -> (R[]) new Object[0]);
```

Comment:
I can understand using Java 9 language features, but I'm not sure what improvement this offers. It reduces the number of lines of code but at the cost of adding a new import and the extra complexity of a lambda. To be sure what this does requires looking up `requireNonNullElseGet`. To me, this change increases complexity and reduces readability without offering any benefit beyond a reduction in LOC.

Reply:
So when I've been reviewing this stuff ^ marked as resolved. I pasted in the java doc into the review comments for people.

Its is a little more advanced reading but I think it is cleaner and with practice people will be able to read it easily. 
like ternary statements.

I WOULD LIKE TO SEE static import of the method to loose `Objects`

Reply:
difference between `requireNonNullElseGet` `requireNonNullElse` I'll let the Author defend that.